File: Data Structures & Algorithms/two-integer-sum/submission-1.go
type IndexCount struct {
    index int
    count int
} 
func twoSum(nums []int, target int) []int {
    hash := make(map[int]IndexCount)
    for index, val := range nums {
        if entry, ok := hash[val]; !ok {
            hash[val] = IndexCount{ index: index, count: 1 }
        } else {
            hash[val] = IndexCount{ index: index, count: entry.count +1}
        }
    }
    for i, val := range nums {
        rest := target - val
        if rest != val && hash[rest].count > 0 {
            return []int{i, hash[rest].index}
        }

        if rest == val && hash[rest].count > 1 {
            return []int{i, hash[rest].index}
        }
    }

    return nil
}

File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        hash = {}
        for num1 in nums:
            if hash.get(num1, 0) > 0:
                return True
            else:
                hash[num1] = hash.get(num1, 0) + 1
        return False